ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/rule-violations/index.tsx

```diff
@@ -118,57 +105,46 @@ const columns = [
   }),
   columnHelper.display({
     id: 'moreInfo',
-    header: () => null,
-    enableGrouping: false,
     size: 50,
-    cell: ({ row }) => (
-      <Tooltip>
-        <TooltipTrigger>
-          <Dialog>
-            <DialogTrigger asChild>
-              <Button variant='outline' size='sm'>
-                <ChevronRight className='h-4 w-4' />
-              </Button>
-            </DialogTrigger>
-            <RuleViolationDetailsComponent details={row.original} />
-          </Dialog>
-        </TooltipTrigger>
-        <TooltipContent>Show the details of the rule violation</TooltipContent>
-      </Tooltip>
-    ),
+    cell: function CellComponent({ row }) {
+      return row.getCanExpand() ? (
+        <Button
+          variant='outline'
+          size='sm'
+          {...{
+            onClick: row.getToggleExpandedHandler(),
+            style: { cursor: 'pointer' },
+          }}
+        >
+          {row.getIsExpanded() ? (
+            <ChevronUp className='h-4 w-4' />
+          ) : (
+            <ChevronDown className='h-4 w-4' />
+          )}
+        </Button>
+      ) : (
+        'No info'
+      );
+    },
+    enableSorting: false,
   }),
 ];
 
-// TODO: This is a temporary solution, which will be replaced with
-// unique subpages to show the rule violation details.
-type Props = {
-  details: RuleViolation;
-};
-const RuleViolationDetailsComponent = ({ details }: Props) => {
+const renderSubComponent = ({ row }: { row: Row<RuleViolation> }) => {
+  const ruleViolation = row.original;
+
   return (
-    <DialogContent
-      className={'max-h-96 overflow-y-scroll text-sm lg:max-w-screen-lg'}
-    >
-      <DialogHeader>
-        <DialogTitle>{details.rule}</DialogTitle>
-        <DialogDescription>{details?.message}</DialogDescription>
-      </DialogHeader>
+    <div className='flex flex-col gap-4'>
+      <div>{ruleViolation.message}</div>
       <div className='grid grid-cols-8 gap-2'>
         <div className='col-span-2 font-semibold'>License:</div>
-        <div className='col-span-6'>{details.license}</div>
+        <div className='col-span-6'>{ruleViolation.license}</div>
         <div className='col-span-2 font-semibold'>License source:</div>
-        <div className='col-span-6'>{details.licenseSource}</div>
+        <div className='col-span-6'>{ruleViolation.licenseSource}</div>
         <div className='col-span-2 font-semibold'>How to fix:</div>
       </div>
-      <MarkdownRenderer markdown={details.howToFix} />
-      <DialogFooter>
-        <DialogClose asChild>
-          <Button type='button' variant='secondary'>
-            Close
-          </Button>
-        </DialogClose>
-      </DialogFooter>
-    </DialogContent>
+      <MarkdownRenderer markdown={ruleViolation.howToFix} />
+    </div>
```

Comment:
Does this expand the width of the table to hide the button if there is a lot of content?

Reply:
It shouldn't, as the how-to-fix is (rendered as) markdown format, whereas for issues, the message was plain string.